src/olympia/ratings/tests/test_views.py

```diff
@@ -1503,6 +1503,60 @@ def test_throttle(self):
             assert str(self.rating.body) == 'I did it'
             assert self.rating.rating == 1
 
+    @override_settings(CACHES=locmem_cache)
+    def test_no_throttling_with_relevant_permission(self):
+        self.client.login_api(self.user)
+
+        with freeze_time('2021-08-09') as frozen_time:
+            for x in range(1, 6):
+                response = self.client.patch(
+                    self.url, {'score': x, 'body': f'Blâh {x}'}
+                )
+                assert response.status_code == 200
+                self.rating.reload()
+                assert str(self.rating.body) == f'Blâh {x}'
+                assert self.rating.rating == x
+
+            # Over the limit now...
+            response = self.client.patch(self.url, {'score': 1, 'body': 'Ooops'})
+            assert response.status_code == 429
+            self.rating.reload()
+            assert str(self.rating.body) == 'Blâh 5'
+            assert self.rating.rating == 5
+
+            # Now with the permission we should be ok.
+            self.grant_permission(self.user, 'Ratings:BypassThrottling')
+            response = self.client.patch(self.url, {'score': 1, 'body': 'Eheheh'})
+            assert response.status_code == 200
+            self.rating.reload()
+            assert str(self.rating.body) == 'Eheheh'
+            assert self.rating.rating == 1
+
+            # Someone else with the same IP would still be stuck.
```

Comment:
Same IP because it is the default in `user_factory()`?

Reply:
Not even related to the user, this is using the (default) client IP. I'll make it more explicit (need to set `REMOTE_ADDR` in the `client.<verb>()` call)